src/windows/osd.c

```diff
@@ -128,3 +134,80 @@ BOOL WINAPI DllMain(HINSTANCE instance, DWORD reason, LPVOID reserved)
 	return TRUE;
 }
 
+int ofi_shm_map(struct util_shm *shm, const char *name, size_t size,
+	int readonly, void **mapped)
+{
+	int ret = FI_SUCCESS;
+	char *fname = 0;
+	size_t len = lstrlen(name) + sizeof(ofi_shm_prefix);
+	LARGE_INTEGER large = {.QuadPart = size};
+	DWORD access = FILE_MAP_READ | (readonly ? 0 : FILE_MAP_WRITE);
+
+	ZeroMemory(shm, sizeof(*shm));
+
+	fname = malloc(len);
+	if (!fname) {
+		ret = -FI_ENOMEM;
+		goto fn_fail;
+	}
+	shm->name = fname;
+
+	lstrcpy(fname, ofi_shm_prefix);
+	lstrcat(fname, name);
+
+	if (!readonly) {
+		shm->shared_fd = CreateFileMapping(INVALID_HANDLE_VALUE, 0,
+			PAGE_READWRITE, large.HighPart, large.LowPart,
+			shm->name);
+		if(!shm->shared_fd) {
+			FI_WARN(&core_prov, FI_LOG_CORE, "CreateFileMapping failed\n");
+			ret = -FI_EINVAL;
+			goto fn_fail;
+		}
+	} else { /* readonly */
+		shm->shared_fd = OpenFileMapping(access, FALSE, shm->name);
+		if(!shm->shared_fd) {
+			FI_WARN(&core_prov, FI_LOG_CORE, "OpenFileMapping failed\n");
+			ret = -FI_EINVAL;
+			goto fn_fail;
+		}
+	}
+
+	shm->ptr = MapViewOfFile(shm->shared_fd, access, 0, 0, size);
+	if (!shm->ptr) {
+		FI_WARN(&core_prov, FI_LOG_CORE, "MapViewOfFile failed\n");
+		ret = -FI_EINVAL;
+		goto fn_fail;
+	}
+
+	*mapped = shm->ptr;
+
+	return ret;
+
+fn_fail:
+	if (shm->shared_fd)
+		CloseHandle(shm->shared_fd);
+	if (fname)
+		free((void*)fname);
+	ZeroMemory(shm, sizeof(*shm));
+	return ret;
+}
+
+int ofi_shm_unmap(struct util_shm *shm)
+{
+	int ret = FI_SUCCESS;
```

Comment:
Why use a variable for the return status if it is never going to change? Just return `FI_SUCCESS`.